Sources/Host/POSIX/PTrace.cpp

```diff
@@ -64,7 +64,13 @@ ErrorCode PTrace::attach(ProcessId pid) {
 
   DS2LOG(Debug, "attaching to pid %" PRIu64, (uint64_t)pid);
 
-  if (wrapPtrace(PTCMD(ATTACH), pid, nullptr, nullptr) < 0)
+#if defined(OS_LINUX) || defined(OS_FREEBSD)
+  auto cmd = PTCMD(ATTACH);
+#elif defined(OS_DARWIN)
+  auto cmd = PT_ATTACHEXC;
+#endif
+
+  if (wrapPtrace(cmd, pid, nullptr, nullptr) < 0)
```

Comment:
Yeah I really need to setup clang-tidy for this.

On Sun, May 8, 2016 at 8:36 PM, kokarez <notifications@github.com> wrote:

> In Sources/Host/POSIX/PTrace.cpp
> <https://github.com/facebook/ds2/pull/327#discussion_r62447668>:
>
> > @@ -64,7 +64,13 @@ ErrorCode PTrace::attach(ProcessId pid) {
> >
> >    DS2LOG(Debug, "attaching to pid %" PRIu64, (uint64_t)pid);
> >
> > -  if (wrapPtrace(PTCMD(ATTACH), pid, nullptr, nullptr) < 0)
> > +#if defined(OS_LINUX) || defined(OS_FREEBSD)
> > +  auto cmd = PTCMD(ATTACH);
> > +#elif defined(OS_DARWIN)
> > +  auto cmd = PT_ATTACHEXC;
> > +#endif
> > +
> > +  if (wrapPtrace(cmd, pid, nullptr, nullptr) < 0)
>
> Not your code -> bracket
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/327/files/942960be6b3e6b91f6076eee6f1ab42bc11e98bd#r62447668>
>



-- 
Stephane Sezer